File: 136. Single Number.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        a = 0
        for i in nums:
            a ^= i
        return a

class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        return 2 * sum(set(nums)) - sum(nums)

class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        dic = {}
        for i in nums:
            p = dic.pop(i, None)
            if not p:
                dic[i] = 1
        return dic.popitem()[0]

File: 136. Single Number.java
class Solution {
    public int singleNumber(int[] nums) {
        HashSet<Integer> seen = new HashSet<>();
        for (int i : nums) {
            if (seen.contains(i)) {
                seen.remove(i);
            } else {
                seen.add(i);
            }
        }
        int ans = 0;
        for (int i : seen) {
            ans = i;
        }
    }
    public int singleNumber(int[] nums) {
        int ans = 0;
        for (int i:nums) {
            ans ^= i;
        }
        return ans;
    }
}